pkg/resources/sources/memory.go

```diff
@@ -116,8 +123,17 @@ func (m *Hugepages) Init(ctx context.Context) (sets.Set[core.ResourceName], erro
 	m.pageSize = hostMem.HugePageSize
 	m.pageCount = hostMem.HugePagesTotal
 
-	m.availableMemory = resource.NewQuantity(int64(m.pageSize*m.pageCount), resource.BinarySI)
-	m.availableHugePages = resource.NewQuantity(int64(m.pageCount), resource.DecimalSI)
+	availableMemory, err := calculateAvailableMemory(MemorySize(m.pageSize*m.pageCount), m.reservedMemorySize)
```

Comment:
it doesn't look good. are you sure it is working? because memory capacity is calculated from hugepages and here you calculate reservedmemory independent.

it looks confusing.

Reply:
memory calculation changed as discussed.